telnyx_rtc/src/main/java/com/telnyx/webrtc/sdk/TelnyxClient.kt

```diff
@@ -184,13 +184,13 @@ class TelnyxClient(
                 // reset audio mode to communication
                 speakerState?.let { setSpeakerMode(it) }
 
-
                 callStateLiveData.postValue(CallState.ACTIVE)
+                callStateLiveData.value = CallState.ACTIVE
```

Comment:
Is this right?

In line `188` we are using postValue, which is thread-safe, and in the next line, we are using .value, which is not thread-safe, and they perform almost the same operation.

Reply:
Yeah they are the same, the issue is  callStateLiveData.postValue(CallState.ACTIVE) doesent set the value somehow (it just updates the observers) and I have to observe the value from the TelnyxClient.kt class might I don't want to do. I could create a separate variablle for CallState.ACTIVE but then there's not single souce of truth.

For thread safety 
`@Override
    public void setValue(T value) {
        super.setValue(value);
    }`
    
    I believe it's just a setter. 
    Now that you mkention though, I think there's a `getState()` method that might have the updated value.

Reply:
Yeah I am actually using it : 
`  if (call.getCallState().value != CallState.ACTIVE)` but somehow it doesn't print the latest value. Let me see if I can make it work

Reply:
IMPORTANT NOTE: I just noticed that if you're in main thread and use postValue it will take a few more millis compared to setValue to dispatch the data. I had this happen when I fill a layout manually, with setValue it was instant but not with post

https://stackoverflow.com/a/51299672/24361182

Reply:
I updated the mutable livedata with a flow and deprecated livedata. This was in Todo for sometime and I think I will  want to deprecate all  mutable livedata infavour of flow because it's threadsafe.